cmd/main.go

```diff
@@ -286,16 +289,45 @@ func ProcessOutChannel(wg *sync.WaitGroup, scConfig *common.SCConfiguration) {
 									log.Infof("post events %s to subscriber %s", d.Address, endPointURI)
 									// make sure event ID is unique
 									event.ID = uuid.New().String()
-									err = restClient.PostCloudEvent(endPointURI, *d.Data)
-									postHandler(err, sub.EndPointURI, d.Address)
+									var status, numSubDeleted int
+									status, err = restClient.PostCloudEvent(endPointURI, *d.Data)
+									if err != nil {
+										log.Errorf("error posting event at %s : %s", endPointURI, err)
+										localmetrics.UpdateEventReceivedCount(d.Address, localmetrics.FAILED)
+
+										// POST return DNSError if server is not reachable
+										var dnsError *net.DNSError
+										// Capture DNS error "lookup consumer-events-subscription-service.cloud-events.svc.cluster.local: no such host"
+										// or timeout error "context deadline exceeded (Client.Timeout exceeded while awaiting headers)"
+										if errors.As(err, &dnsError) || os.IsTimeout(err) {
+											// has subscriber failed to connect for 10 times delete the subscribers
+											if scConfig.SubscriberAPI.IncFailCountToFail(clientID) {
```

Comment:
do you dues the threshold to wait few tries before cleaning up ?
log.Errorf("client %s not responding, waiting %d times before marking to delete subscriber",
						clientAddress, h.subscriberAPI.FailCountThreshold()-h.subscriberAPI.GetFailCount(clientID))